tests/Unit/Evolution/DiscontinuousGalerkin/Test_LiftFromBoundary.cpp

```diff
@@ -0,0 +1,259 @@
+      const auto lower_face_mesh =
+          volume_mesh.slice_away(lower_direction.dimension());
+      const auto lower_face_logical_coords =
+          interface_logical_coordinates(lower_face_mesh, lower_direction);
+      const auto lower_face_inertial_coords = map(lower_face_logical_coords);
+      const auto lower_face_det_jacobian =
+          determinant(map.jacobian(lower_face_logical_coords));
+      const auto lower_face_normal =
+          unnormalized_face_normal(lower_face_mesh, map, lower_direction);
+      const auto lower_face_normal_magnitude = magnitude(lower_face_normal);
+      auto lower_unit_face_normal = lower_face_normal;
+      for (auto& t : lower_unit_face_normal) {
+        t /= get(lower_face_normal_magnitude);
+      }
+
+      const auto lower_boundary_corrections = normal_dot_flux<tags>(
+          lower_unit_face_normal,
+          polynomial_volume_fluxes<tags>(lower_face_inertial_coords, powers));
+
+      // Now perform lifting
+      evolution::dg::lift_boundary_terms_gauss_points(
+          make_not_null(&dt_vars_lifted_two_at_a_time), volume_det_inv_jacobian,
+          volume_mesh, direction.dimension(), boundary_corrections,
+          face_normal_magnitude, face_det_jacobian, lower_boundary_corrections,
+          lower_face_normal_magnitude, lower_face_det_jacobian);
+    }
+  }
+
+  Variables<db::wrap_tags_in<
+      Tags::div, typename std::decay_t<decltype(volume_fluxes)>::tags_list>>
+      div_fluxes{volume_mesh.number_of_grid_points()};
+  divergence(make_not_null(&div_fluxes), volume_fluxes, volume_mesh,
+             volume_inv_jacobian);
+  Variables<db::wrap_tags_in<Tags::dt, tags>> expected_dt_vars{
+      volume_mesh.number_of_grid_points()};
+  std::copy(div_fluxes.data(), div_fluxes.data() + div_fluxes.size(),
+            expected_dt_vars.data());
+  expected_dt_vars *= -1.0;
+
+  Approx local_approx = Approx::custom().epsilon(eps).scale(1.0);
+  tmpl::for_each<typename decltype(expected_dt_vars)::tags_list>(
+      [&dt_vars_lifted_one_at_a_time, &dt_vars_lifted_two_at_a_time,
+       &expected_dt_vars, &local_approx](auto tag_v) noexcept {
+        using tag = tmpl::type_from<decltype(tag_v)>;
+        CHECK_ITERABLE_CUSTOM_APPROX(get<tag>(dt_vars_lifted_one_at_a_time),
+                                     get<tag>(expected_dt_vars), local_approx);
+        CHECK_ITERABLE_CUSTOM_APPROX(get<tag>(dt_vars_lifted_two_at_a_time),
+                                     get<tag>(expected_dt_vars), local_approx);
+      });
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.DG.LiftFromBoundary", "[Unit][Evolution]") {
+  // We need larger epsilon in 2d and 3d because the lifting does not work so
+  // well with the curved meshes. The reason isn't clear. Projecting
+  // J d xi^{hat{i}} /dx^i F^i to the boundary and then lifting it back does not
+  // make it so that the metric identities are satisfied discretely in the weak
+  // form.
```

Comment:
Please make this comment and the one on lines 118–123 consistent (perhaps by deleting one of them?). The one on 118-123 says "we might want to do X" and the one here says "X doesn't do the thing".